tendrl/notifier/notification/handlers/snmp_handler.py

```diff
@@ -0,0 +1,251 @@
+from pysnmp.hlapi import usmHMACMD5AuthProtocol
+from pysnmp.hlapi import UsmUserData
+from tendrl.commons.config import load_config
+from tendrl.commons.event import Event
+from tendrl.commons.message import ExceptionMessage
+from tendrl.commons.utils.log_utils import log
+from tendrl.notifier.notification import NotificationPlugin
+
+# USM users fixed snmp engine id
+ENGINE_ID = "8000000001020304"
+# default snmp port
+PORT = 162
+V2_ENDPOINT = "v2_endpoint"
+V3_ENDPOINT = "v3_endpoint"
+ALERT_TYPES = {0: "STATUS",
+               1: "UTILIZATION"}
+
+
+class SnmpEndpoint(object):
+    # Base class for SNMP endpoint (credential and host+port pair)
+
+    def __init__(self, host_ip="127.0.0.1", proto=0):
+        self.proto = proto
+        self.host = host_ip
+        self.port = PORT
+        self.engineid = ENGINE_ID
+
+
+class V2Endpoint(SnmpEndpoint):
+    # Class encapsulating an SNMPv2c community, host+port pair
+    # 2 represent SNMPv2
+    def __init__(self, host_ip, community):
+        super(V2Endpoint, self).__init__(host_ip, 2)
+        self.community = community
+
+
+class V3Endpoint(SnmpEndpoint):
+    # Class encapsulating an SNMPv2c community, host+port pair
+    # 2 represent SNMPv2
+    def __init__(self, host_ip, username, auth_key, priv_key):
+        super(V3Endpoint, self).__init__(host_ip, 3)
+        self.usm_user = UsmUserData(userName=username,
+                                    authKey=auth_key,
+                                    privKey=priv_key,
+                                    authProtocol=usmHMACMD5AuthProtocol,
+                                    privProtocol=usmDESPrivProtocol
+                                    )
+
+
+class SnmpHandler(NotificationPlugin):
+
+    def __init__(self):
+        self.name = 'snmp'
+        self.user_configs = []
+
+    def set_destinations(self):
+        try:
+            self.user_configs = self.get_alert_destinations()
+        except (
+            EtcdException,
+            EtcdKeyNotFound,
+            ValueError,
+            KeyError,
+            SyntaxError
+        ) as ex:
+            if type(ex) != EtcdKeyNotFound:
+                self.user_configs = []
+                return
+            raise ex
+
+    def get_alert_destinations(self):
+        snmp_config = load_config(
+            'notifier',
+            '/etc/tendrl/notifier/snmp.conf.yaml'
+        )
+        return snmp_config
+
+    def format_message(self, alert):
+        resource = alert.resource.replace("_", " ").title()
+        severity = alert.severity
+        message = alert.tags['message']
+        if alert.alert_type == ALERT_TYPES[0]:
+            suffix = "status changed"
+        elif alert.alert_type == ALERT_TYPES[1]:
+            suffix = "threshold breached"
+        return "[Tendrl Alert] %s, %s: %s-%s" % (
+            resource, severity, suffix, message)
+
+    def getPDU(self, message):
+        pdu = [
+            ObjectType(ObjectIdentity('1.3.6.1.4.2312.19.1.1'),
```

Comment:
Add a comment to explain about the number passed here.

Reply:
ok